tsl/src/compression/compression.c

```diff
@@ -2376,12 +2642,29 @@ decompress_batches_for_update_delete(List *chunks, List *predicates)
 			scankeys =
 				build_update_delete_scankeys(&decompressor, filters, &num_scankeys, &null_columns);
 		}
-		if (decompress_batches(&decompressor,
-							   scankeys,
-							   num_scankeys,
-							   null_columns,
-							   is_null,
-							   &chunk_status_changed))
+		if (index_filters)
+		{
+			index_scankeys = build_update_delete_index_scankeys(&decompressor,
+																index_filters,
+																&num_index_scankeys,
+																estate);
+			if (num_index_scankeys && decompress_batches_using_index(&decompressor,
+																	 index_scankeys,
+																	 num_index_scankeys,
+																	 scankeys,
+																	 num_scankeys,
+																	 &chunk_status_changed))
+			{
+				if (chunk_status_changed == true)
+					ts_chunk_set_partial(chunk);
+			}
```

Comment:
This seems duplicated for index and no-index decompression. We should refactor to have a single place where we change the chunk status.

Reply:
Fixed.